.github/workflows/iic2513-node.yml

```
@@ -4,21 +4,42 @@ on: pull_request
 
 jobs:
   tests:
+    name: Tests and linters
+    timeout-minutes: 10
     runs-on: ubuntu-latest
-
-    timeout-minutes: 12
-
+    env:
+      DATABASE_URL: postgresql://localhost/postgres
+      DB_DIALECT: postgres
+      DB_USERNAME: postgres
+      DB_PASSWORD: password
+    services:
+      db:
+        env:
+          POSTGRES_HOST_AUTH_METHOD: trust
+        image: postgres:12.4
+        ports: ['5432:5432']
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+    
     strategy:
       matrix:
         node-version: [ 12.x ]
-
     steps:
-    - uses: actions/checkout@v2
-    - name: Use Node.js ${{ matrix.node-version }}
-      uses: actions/setup-node@v2
-      with:
-        node-version: ${{ matrix.node-version }}
-    - name: Install dependencies
-      run: yarn install
-    - name: Run ESLint
-      run: yarn lint
+      - uses: actions/checkout@v2
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: Install dependencies
+        run: yarn install
+      - name: Setup Database
+        run: |
+          NODE_ENV=test yarn sequelize db:create
+          NODE_ENV=test yarn sequelize db:migrate
+      - name: Run ESLint
+        run: yarn lint
+      - name: Run jest tests
+        run: yarn test --silent
```

@B-Dominguez Viendo el output de la ejecución de los tests, muchas rutas están respondiendo con `401` cuando no deberían hacerlo. Me di cuenta de que es porque en #26 se utilizan rutas protegidas, que requieren el uso de JWT, y para eso estamos utilizando una env variable llamada `JWT_SECRET`.

La solución simple sería agregar la env variable a este comando, pero tal vez sería bueno explorar una forma más ordenada de definir variables de ambiente, ya que también estamos utilizando `NODE_ENV`. ¿Existirá por ejemplo una forma de definir un archivo `.env.test` aquí mismo y escribir las env variables ahí?

```suggestion
        run: JWT_SECRET=some-secret yarn test --silent
```

---

Genial, esto efectivamente arregló el problema y la action ya pasó.

Sobre cómo definir las variables env, se hace de hecho bajo la llave `jobs.JOB.env` en el `ACTION.yml`. Para `NODE_ENV` me había hecho ruido ponerla ahí siendo que necesitaba solo para los comandos de la db, pero decantándolo mejor, el job por algo se llama `tests`, y considerando además lo de que ya son dos casos de uso de variables de entorno, concuerdo con que es mejor definirlas ahí (ya lo hice en el último _commit_).